### PR TITLE
Single Step Shell Code

### PR DESCRIPTION
**Pull request by chirag-compro**

### CODE REVIEW

File: client/app/homepage/homepage.module.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export const routes = [
+  { path: '', component: HomePage, pathMatch: 'full' }
+];
+
+@NgModule({
+  imports: [ CommonModule, RouterModule.forChild(routes), FormsModule ],
+  declarations: [ HomePage ]
+})
+export class HomePageModule {
+  static routes = routes;
```
Comment (vsingb3) [inline client/app/homepage/homepage.module.ts]: static routes = routes;
Why is this required ?

File: client/app/layout/layout.component.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
+import { Component, ViewEncapsulation, OnInit, AfterViewInit, ElementRef, Output, EventEmitter } from '@angular/core';
```
Comment (ashish-compro) [inline client/app/layout/layout.component.ts]: Remove View Encapsulation as it has not been used in the component.

File: client/app/layout/layout.component.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    // this.$sidebar.find('.active .active').closest('.collapse').collapse('show')
+    //   .siblings('[data-toggle=collapse]').removeClass('collapsed');
+  }
+
+  collapseNavigation(): void {
+    // this method only makes sense for non-static navigation state
+    if (this.isNavigationStatic()
+      && (this.configFn.isScreen('lg') || this.configFn.isScreen('xl'))) { return; }
+
+    jQuery('layout').addClass('nav-collapsed');
```
Comment (ashish-compro) [inline client/app/layout/layout.component.ts]: Can we find a way to code this line in native javascript instead of using jquery functions ?

File: client/app/homepage/homepage.module.ts (hunk 1 above)
Comment (chirag-compro) [reply client/app/homepage/homepage.module.ts]: This can be removed. It is required in the case where the routes have to be configured in the module class and is put static so that all the instances of this modulle use the same route properties

File: client/app/home/home.module.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+
+import { NgModule }      from '@angular/core';
+import { CommonModule }  from '@angular/common';
+import { FormsModule }  from '@angular/forms';
+import { TooltipModule } from 'ng2-bootstrap';
```
Comment (ashish-compro) [inline client/app/home/home.module.ts]: It should be removed as it has not been used in the home module.

File: client/app/home/layout.component.ts [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+      // return for bubbled events
+      if (e.target !== e.currentTarget) { return; }
+
+      jQuery(this).closest('li').removeClass('open');
+    });
+  }
+  ngAfterViewInit(): void{
+    this.url = this.router.url;
+    if(this.url == "/app/homepage")
+    this.showSearchForm = true;
```
Comment (ashish-compro) [inline client/app/home/layout.component.ts]: Please see to this code block. Not sure if the if-else block is required.

File: client/app/task-builder/task-builder.component.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+initialiseTaskData() {
+   	this.route.data
+                     .subscribe(
+                       taskData => this.TaskData = taskData,
+                       error =>  this.errorMessage = <any>error);
+					   this.TaskData = this.TaskData["taskData"];
+					   this.StepData = this.TaskData["stepData"];
+					   this.AppImage = this.ApplicationImage(this.TaskData["app"]);
+  }
+ ApplicationImage (taskApp){
```
Comment (ashish-compro) [inline client/app/task-builder/task-builder.component.ts]: one better way of writing this switch case could be - 
function(taskApp){
return {
     "Excel": "assets/images/Excel.png"
}[taskApp]
}

**Review by vsingb3 — APPROVED** [review]
